File: src/main/java/template_method/PostAccountReqBody.java
package template_method;

import template_method.exception.BadRequestException;

public record PostAccountReqBody(String accountNumber, String branchCode) implements RequestBody {
    public PostAccountReqBody {
        if (accountNumber.length() != 5) {
            throw new BadRequestException("invalid account Number");
        }

        if (branchCode.length() != 3) {
            throw new BadRequestException("invalid branch code");
        }
    }
}

File: kotlin/src/main/kotlin/template_method/PostAccountReqBody.kt
package template_method

import template_method.exception.BadRequestException

data class PostAccountReqBody(val accountNumber: String, val branchCode: String) : RequestBody {
    init {
        if (accountNumber.length != 5) {
            throw BadRequestException("invalid account Number")
        }

        if (branchCode.length != 3) {
            throw BadRequestException("invalid branch code")
        }
    }
}
